airbyte-ci/connectors/pipelines/pipelines/helpers/changelog.py

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+import re
+from operator import attrgetter
+from pathlib import Path
+
+import semver
+from typing_extensions import Set
+
+
+class Changelog:
+    def __init__(self, changelog_file_path: Path) -> None:
+        self.changelog_file_path = changelog_file_path
+        if not changelog_file_path.exists():
+            raise Exception(f"File {changelog_file_path} does not exist")
+        if not changelog_file_path.is_file():
+            raise Exception(f"File {changelog_file_path} is not a regular file")
+        self.original_markdown_lines = changelog_file_path.read_text().splitlines()
```

Comment:
Can we pass the `original_markdown_lines` as a constructor parameter and avoid path checks logic there?
I'd love to avoid direct filesystem access if possible.
I'd prefer the changelog read to happen from a dagger directory object. This will help us make this Changelog agnostic of the underlying filesystem / path: a dagger directory could come from a container, a remote git repo or the host fs.

Feel free to not change how the documentation file path is fetched, but please change the constructor to take the raw markdown string as an input instead of the file path